### PR TITLE
integrate the entry for location mismatch error

### PR DESCRIPTION
**Pull request by yanshil**
os.path.realpath and pathlib.resolve() have different behavior in py3.7 and py3.7+

In python-3.7, os.path.realpath(path) return the drive location like `l:\opt\rez\2.103.4\platform-windows`
But in python-3.8, it will return the unc path like `\\\\studio-dfs\\studio\\...`

I still came across with location mismatched error in some `net use` drive in py3.8 and py3.9 and found out that the location mismatch error in rez.package_repository was not fixed

```
   File "l:\opt\rez\2.103.4\platform-windows\python-3.9\lib\site-packages\rez\utils\data_utils.py", line 566, in func
     value = getattr(self.wrapped, key, None)
   File "l:\opt\rez\2.103.4\platform-windows\python-3.9\lib\site-packages\rez\utils\data_utils.py", line 260, in __get__
     result = self.func(instance)
   File "l:\opt\rez\2.103.4\platform-windows\python-3.9\lib\site-packages\rez\package_resources.py", line 374, in root
     return self._root()
   File "l:\opt\rez\2.103.4\platform-windows\python-3.9\lib\site-packages\rez\package_resources.py", line 511, in _root
     if self.base is None:
   File "l:\opt\rez\2.103.4\platform-windows\python-3.9\lib\site-packages\rez\utils\data_utils.py", line 566, in func
     value = getattr(self.wrapped, key, None)
   File "l:\opt\rez\2.103.4\platform-windows\python-3.9\lib\site-packages\rez\package_resources.py", line 535, in wrapped
     return self.parent
   File "l:\opt\rez\2.103.4\platform-windows\python-3.9\lib\site-packages\rez\utils\data_utils.py", line 260, in __get__
     result = self.func(instance)
   File "l:\opt\rez\2.103.4\platform-windows\python-3.9\lib\site-packages\rezplugins\package_repository\filesystem.py", line 285, in parent
     package = self._repository.get_resource(
   File "l:\opt\rez\2.103.4\platform-windows\python-3.9\lib\site-packages\rez\package_repository.py", line 453, in get_resource
     handle = self.make_resource_handle(resource_key, **variables)
  File "l:\opt\rez\2.103.4\platform-windows\python-3.9\lib\site-packages\rez\package_repository.py", line 431, in make_resource_handle
    raise ResourceError("location mismatch - requested %r, repository "
rez.exceptions.ResourceError: location mismatch - requested 'l:\\apps\\studio\\prod', repository location is '\\\\studio-dfs\\studio\\apps\\studio\\prod'
```

### CODE REVIEW

**Comment by JeanChristopheMorinPerso** [comment]
Hello @yanshil , can you please sign the CLA? Signing the CLA means that you accept the terms of Rez' license. If you do this change on behalf of your employer, I suggest to talk to them before signing the CLA.

Thanks and let us know if you have questions.

**Comment by yanshil** [comment]
@JeanChristopheMorinPerso CLA and DCO fixed. Thanks!

File: src/rez/utils/filesystem.py [hunk 1]
```diff
@@ -697,3 +698,22 @@ def windows_long_path(dos_path):
         path = "\\\\?\\" + path
 
     return path
+
+
+def compare_location(loc1, loc2):
```
Comment (JeanChristopheMorinPerso) [inline src/rez/utils/filesystem.py:703]: This function is kind of opinionated. Both paths are lower cased but only the first path is normalized (through canonical_path). Also the exception mentions "repository".

So to have a function in src/rez/utils/filesystem.py is alright, but the function would need to be generic and non opinionated. ALso I'd name it `compare_paths`.

Additionally, a unit test would be welcome.

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

**Pull request closed without merging**